lib/features/feedback/cubit/send_feedback_cubit.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:bloc/bloc.dart';
+import 'package:fluttercon/common/repository/api_repository.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'send_feedback_state.dart';
+part 'send_feedback_cubit.freezed.dart';
+
+class SendFeedbackCubit extends Cubit<SendFeedbackState> {
+  SendFeedbackCubit({
+    required ApiRepository apiRepository,
+  })  : _apiRepository = apiRepository,
+        super(const SendFeedbackState.initial());
+
+  final ApiRepository _apiRepository;
+
+  Future<void> sendFeedback({
+    required String feedback,
+    int? index,
+    int? rating,
+  }) async {
+    try {
+      emit(SendFeedbackState.loading(index: index));
+      final message = await _apiRepository.submitFeedback(
```

Comment:
You want to avoid passing multiple values across levels of the app. Usually what this leads to is, when the value changes in one place, you have to change function parameters in multiple places. 

Instead what you can do is use the DTO approach. For instance, in this case you can have this as follows. 

- In `lib/common/data/models` make a new `freezed` file: `feedback_dto`
```dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'feedback_dto.freezed.dart';
part 'feedback_dto.g.dart';

@freezed
class FeedbackDTO with _$FeedbackDTO {
  factory FeedbackDTO({
    required String feedback,
    int? rating,
  }) = _FeedbackDTO;
	
  factory FeedbackDTO.fromJson(Map<String, dynamic> json) =>
			_$FeedbackDTOFromJson(json);
}
```
- Run code gen
- In `lib/common/repository/api_repository.dart:140` change the signature of the function to 
```dart
  Future<String> submitFeedback({
    required FeedbackDTO feedbackDTO
  })
```
- In `lib/common/repository/api_repository.dart:147-150`
```dart
 body: feedbackDTO.toJson(),
```
- Then lastly in this file, 
```dart
_apiRepository.submitFeedback(feedbackDTO: FeedbackDTO(
  feedback: feedback,
  rating: rating,
));
```

In this manner, if the attributes on the API changes, you don't change the signature of the function in the repository, it just changes in the cubit. Looks not very consequential at this stage, but really helpful say when you have an API request that has many parameters, you don't want to be passing them one by one across several layers.

Reply:
Noted. Thanks